File: E-Tour/Java/Backend/E-Tour/src/main/java/com/example/controller/CustomerController.java
package com.example.controller;

import com.example.models.Customer;
import com.example.services.CustomerServices;

import java.util.HashMap;
import java.util.Map;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;


@RestController
@RequestMapping("/api/customers")
@CrossOrigin(origins = "http://localhost:3000") 
public class CustomerController {

    @Autowired
    private CustomerServices customerServices;

    // Register new customer
    @PostMapping("/register")
    public ResponseEntity<?> registerCustomer(@RequestBody Customer customer) {
        Customer savedCustomer = customerServices.saveCustomer(customer);
        return ResponseEntity.status(HttpStatus.CREATED).body(savedCustomer);
    }

//    // Get customer by ID
//    @GetMapping("/{id}")
//    public Optional<Customer> getCustomerById(@PathVariable Long id) {
//        return customerServices.getCustomerById(id);
//    }
    // Get customer details by email after login
    @GetMapping("/check/{email}")
    public ResponseEntity<Customer> getCustomerDetails(@PathVariable String email) {
        try {
            Customer customer = customerServices.getCustomerByEmail(email);
            System.out.println("Inside check");
            if (customer != null) {
            	System.out.println("Inside not null");
            	System.out.println("customer"+customer.getFirstName());
                return ResponseEntity.ok(customer);
            } else {
            	System.out.println("Inside null");
                return ResponseEntity.notFound().build();
            }
        } catch (Exception e) {
            return ResponseEntity.status(HttpStatus.INTERNAL_SERVER_ERROR).build();
        }
    }

    // Get customer by email for login
    @PostMapping("/login")
    public ResponseEntity<Map<String, String>> loginCustomer(@RequestBody Customer customer) {
        System.out.println(customer);
        Customer found = customerServices.getCustomerByEmailandPassword(customer);

        if (found!=null) {
            Map<String, String> response = new HashMap<>();
            response.put("message", "Login Successful");
            return ResponseEntity.ok(response);
        }

        Map<String, String> errorResponse = new HashMap<>();
        errorResponse.put("message", "Invalid email or password.");
        return ResponseEntity.status(HttpStatus.UNAUTHORIZED).body(errorResponse);
    }

}

File: Controllers/CustomerController.cs
using E_Tour.Models;
using Etour.DTO;
using Etour.Service;
using Microsoft.AspNetCore.Cors;
using Microsoft.AspNetCore.Mvc;

namespace Etour.Controllers
{
        [ApiController]
        [Route("api/customers")]
        [EnableCors("AllowReactApp")]

    public class CustomerController : Controller
    {
        
        private readonly ICustomerService _customerService;

        public CustomerController(ICustomerService customerService) // Constructor Injection
        {
            _customerService = customerService;
        }

        [HttpPost("register")]
        public async Task<ActionResult<Customer>> RegisterCustomer([FromBody] Customer customer)
        {
            Console.WriteLine("inside the Register customer");
            var result = await _customerService.SaveCustomer(customer);
            return Ok(result);
        }

        [HttpGet("check/{email}")]
        public async Task<ActionResult<Customer>> GetCustomerDetails(string email)
        {
            try
            {
                var customer = await _customerService.GetCustomerByEmail(email);
                if (customer != null)
                {
                    return Ok(customer);
                }
                return NotFound();
            }
            catch (Exception)
            {
                return StatusCode(500, "Internal Server Error");
            }
        }


        [HttpPost("login")]
        public async Task<ActionResult<Dictionary<string, string>>> LoginCustomer([FromBody] UserDto userdto)
        {
            var found = await _customerService.GetCustomerByEmailAndPassword(userdto);

            if (found != null)
            {
                return Ok(new Dictionary<string, string>
                {
                    { "message", "Login Successful" }
                });
            }

            return Unauthorized(new Dictionary<string, string>
            {
                { "message", "Invalid email or password." }
            });
        }


    }
}
